libcst/metadata/base_provider.py

```diff
@@ -44,9 +44,17 @@ class BaseMetadataProvider(MetadataDependent, Generic[_ProvidedMetadataT]):
     # explanation.
     _computed: MutableMapping["CSTNode", _ProvidedMetadataT]
 
-    def __init__(self) -> None:
+    is_cache_required: bool = False
+
+    def __init__(self, cache: object = None) -> None:
         super().__init__()
         self._computed = {}
+        if self.is_cache_required and not cache:
+            # The metadata provider implementation is responsible to store and use cache.
+            raise Exception(
+                f"Cache is required for initializing {self.__class__.__name__}."
```

Comment:
This exception can explain *how* to use cache, such as "Cache is required..... Please use <method name here> to calculate cache and pass in using the `cache` parameter." Something to the effect of helping the user to the solution instead of just saying there's a problem.

Reply:
Make sense. We don't have the helper to get cache yet, so I didn't add it in the comment. I can add it now as a placeholder.